docs/configuration.asciidoc

```diff
@@ -0,0 +1,131 @@
+////
+Goal of this doc:
+Provide a complete reference of all available configuration options and where/how they can be set. (Any Elastic-specific configuration options are listed directly. General OpenTelemetry configuration options are linked.)
+
+Assumptions we're comfortable making about the reader:
+* They are familiar with Elastic
+* They are familiar with OpenTelemetry
+////
+
+[[configuration]]
+== Configure
+
+:language: Java
+:language_lc: java
+:distro_name: Elastic Distribution for OpenTelemetry {language}
+
+include::release-status.asciidoc[]
+
+////
+TO DO: How do users set configuration options?
+////
+[discrete]
+[[configure-methods]]
+=== Configuration methods
+
+////
+TO DO:
+Are there multiple ways to set configuration options?
+For example, can they use environment variables OR a config file?
+If there are, what are they and why would you use each method?
+Where/how do you use each method?
+Provide an example to illustrate the syntax.
+////
+
+////
+TO DO:
+If configuration options are defined using multiple methods,
+which one takes precedence over the other(s)?
+////
+
+[discrete]
+[[configuration-options]]
+=== Configuration options
+
+Because the Elastic Distribution for OpenTelemetry Java ("the distro") is an extension of the https://github.com/open-telemetry/opentelemetry-java-instrumentation[OpenTelemetry Java agent], it supports both:
+
+* General OpenTelemetry SDK configuration options
+* Elastic-specific configuration options that are only available when using the distro
+
+[discrete]
+[[configure-otel-sdk-options]]
+==== OpenTelemetry SDK configuration options
+
+////
+TO DO:
+Is this true?
+Are there any options that aren't supported?
+Any options that shouldn't be used?
+////
+The distro supports all configuration options listed in the https://opentelemetry.io/docs/languages/sdk-configuration/general/[OpenTelemetry General SDK Configuration documentation].
+
+////
+TO DO:
+Are there Elastic-specific custom configuration options
+in addition to the general OpenTelemetry SDK
+configuration options? Or are all the customizations
+"behind the scenes"?
+If not, you can delete this section.
+////
```

Comment:
There are elastic specific configuration options, these are listed in the extension READMEs, and are changing quite quickly because we contribute them upstream and then replace the extension with the contribution which means they become otel options.

I think best to leave this as a to do and open a separate issue for this to be updated later

But worth mentioning here that we have 3 sets of options:
1. upstream vanilla agent that this is a distro of
2. upstream contributions that vanilla agent users can optionally add in but that we are including in our distro (so our agent supports *extra* OTEL_... options that the upstream vanilla agent doesn't
3. ELASTIC_OTEL_... options that we add in our distro

Reply:
Thanks. This is helpful. Would this language be accurate?

>### OpenTelemetry SDK configuration options
>
>The Elastic Distribution for OpenTelemetry Java supports all configuration options listed in the general [OpenTelemetry SDK configuration documentation](https://opentelemetry.io/docs/languages/sdk-configuration/general/).
>
>### Elastic distro configuration options
>
>In addition to general OpenTelemetry SDK configuration options, there are two kinds of configuration options that are only available in the Elastic Distribution for OpenTelemetry Java:
>
>* Additional `OTEL_*` options that Elastic plans to or contribute upstream to the [OpenTelemetry SDK for Java](https://opentelemetry.io/docs/languages/java), but are not yet available upstream.
>* `ELASTIC_OTEL_*` options that are specific to Elastic and will always live in the Elastic Distribution for OpenTelemetry Java (in other words, it will _not_ be added upstream).
>
>Read more about configuration options that are only available in the distro in [the elastic/elastic-otel-java repo's README](https://github.com/elastic/elastic-otel-java?tab=readme-ov-file#features).